models/lido/accounting/ethereum/lido_accounting_ethereum_deposits.sql

```diff
@@ -12,10 +12,15 @@
 --https://dune.com/queries/2011901
 --ref{{'lido_accounting_ethereum_deposits'}}
 
-	SELECT 
-        block_time AS period,
-        amount_staked*POWER(10,18) AS amount_staked,
-        LOWER('0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2') AS token, --ETH
-        tx_hash
-    FROM {{ ref('staking_ethereum_deposits') }} 
-    WHERE depositor_entity = 'Lido'
\ No newline at end of file
+	SELECT  block_time as period, 
+        sum(cast(value as DOUBLE)) as amount_staked, 
+        LOWER('0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2') AS token,
+        tx_hash,
+        date_trunc('day', block_time) as day 
+        FROM  {{source('ethereum','traces')}} 
```

Comment:
with the switch to reading from a base table source, this spell starts to make me think it would benefit from being an incremental table. what do ya think?

more frequent refreshes, more efficient each run (only read one week of data from traces), etc.

iirc, most lido spells aren't incremental, so i don't know if that'd throw off the latency between all the different spells. if anything, this one spell would be every hour or so

Reply:
thanks for review @jeff-dude, I agree, would like to have more fresh data in this spell

Reply:
you'll need to include two more things:
1 - unique keys property in config block (plenty of examples in other spells). these will be used for the merge statement which dbt builds to help determine insert or update on incoming records
2 - a corresponding test on unique keys, placed in the schema yml file which hosts this spell. it's a "dbt unique combo of columns" test. again, lots of examples in other schema files

Reply:
What would be better, have only hash in unique key or hash and txn date? I see that in other spells were choosen 2 fields also if logically hash is enough as unique key

Reply:
`block_date` can be a ideal column for `partition_by`.
If `tx_hash` is logically enough for `unique_key`, you can have only `tx_hash` in unique_key.

Reply:
yep, feel free to keep the minimum columns needed